### PR TITLE
codebase cleanup

### PR DESCRIPTION
**Pull request by tianyu-l**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #228

This PR makes some cleanups before release:
1. use realistic settings for 13B and 70B model training
2. remove miniPile dataset as it is skewed and we don't support randomized data loading yet
3. make naming, comments, logging consistent

### CODE REVIEW

File: train_configs/llama_13b.toml [hunk 1]
```diff
@@ -25,14 +24,13 @@ name = "AdamW"
...
+batch_size = 16
 seq_len = 4096
-warmup_steps = 200  # lr scheduler warm up
+warmup_steps = 200  # lr scheduler warm up, normally 20% of the train steps
 max_norm = 1.0  # grad norm clipping
 steps = 1000
 data_parallel_degree = -1
-# 8-way TP, adjust to 2/4 for local(single host) runs
-tensor_parallel_degree = 8
+tensor_parallel_degree = 8  # 8-way TP, adjust to 2/4 for local (single host) runs
```
Comment (wanchaol) [inline train_configs/llama_13b.toml]: Let's maybe disable TP for 13b model runs? we can have:
1. 13b FSDP only
2. 70B FSDP + TP

File: torchtitan/datasets/hf_datasets.py [hunk 2]
```diff
@@ -14,7 +14,6 @@
 
 _supported_datasets = {
     "alpaca": "tatsu-lab/alpaca",
-    "minipile": "JeanKaddour/minipile",
```
Comment (gnadathur) [inline torchtitan/datasets/hf_datasets.py:17]: Should we remove the support totally or add a comment on the limitations ?

File: train_configs/llama_13b.toml (hunk 1 above)
Comment (tianyu-l) [reply train_configs/llama_13b.toml]: In that case what hardware should we assume, e.g. 64GPU?
batch size and AC option and even learning rate will depend on that...

Comment (tianyu-l) [reply train_configs/llama_13b.toml]: resolved in offline discussion

File: torchtitan/datasets/hf_datasets.py (hunk 2 above)
Comment (tianyu-l) [reply torchtitan/datasets/hf_datasets.py:17]: resolved in offline discussion
